File: ruby/lesson 7/graph.rb
class Graph
    def initialize(capacity)
        @adj = Array.new(capacity) { [] }
    end
    
    def size
        @adj.length
    end
    
    def get(v)
        @adj[v]
    end
    
    def add(v, w)
        @adj[v].push(w)
        @adj[w].push(v)
    end
end

File: csharp/lesson 7/Graph.cs
public class Graph
{
    private IList<int>[] adj;
    public int Size { get => adj.Length; }
    
    public Graph(int capacity) 
    {
        adj = new IList<int>[capacity];
        for (var i = 0; i < capacity; i++)
        {
            adj[i] = new List<int>();
        }
    }

    public void Add(int v, int w)
    {
        adj[v].Add(w);
        adj[w].Add(v);
    }

    public IList<int> Get(int v) 
    {
        return adj[v];
    }
}
